pkg/secret-store/service/k8s_secrets_service.go

```diff
@@ -132,20 +131,26 @@ func (secretsService *K8sSecretsService) DeleteSecret(name string) error {
 }
 
 func (secretsService *K8sSecretsService) UpdateSecret(secret models.Secret) (*models.Secret, error) {
-	entity := secretstore.SecretEntity{}
-	name := *secret.Name
-
-	err := secretsService.EntityStore.Get(secretsService.OrgID, name, &entity)
-
-	if err == store.ErrKeyNotFound {
-		return nil, SecretNotFound{}
+	name := secret.Name
+	nameFilter := []entitystore.FilterStat{
+		entitystore.FilterStat{
+			Subject: "Name", Verb: entitystore.FilterVerbEqual, Object: name,
+		},
 	}
 
+	var entities []*secretstore.SecretEntity
+	err := secretsService.EntityStore.List(secretsService.OrgID, nameFilter, &entities)
```

Comment:
Why are you calling list as opposed to Get?

Reply:
The previous logic relied on checking an error that was only thrown by libkv to determine if the entity was not found. That error is not thrown by postgres. This implementation allows me to determine if the entity was not found without relying on those errors.

Reply:
What does Get return then? The point is, that it sounds like you are highlighting a bug in the entity store.  If this is the case, then we need to fix it, not just work around it by using List everywhere.  This will be an issue everywhere we use Get, so we should just fix it.